### PR TITLE
Notes flow v0

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
screenflows

- Happy path works but some TODOs sprinkled around
- Deeplinking also works with just `ephemeralOwner` for the sender to open the note and revert (so the app is stateless for the sender as that can be fetched in history feed by indexer)

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/NoteScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  };
+
+  const { status, message, exec } = useSendAsync(
+    account.enclaveKeyName,
+    createTxHandle(ephemeralNote.owner, ephemeralSignature)
+  );
+
+  // TODO: load estimated fees
+  // NOTE: These fees seem difficult to communicate to user, think about it.
+  const fees = 0.05;
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/NoteScreen.tsx:149]: re: communicating to user, my current best guess is
- if fee<$0.10, subtle, gray indicator under the send button that shows either (total you're paying) or (net you're receiving) after fees
- fee >= $0.10, indicator is still small but now a ⚠️ yellow caution indicator
- fee > $1.00 or simulation unavailable, ❗️red warning, send button disabled or behind a confirm popup

Comment (dcposch) [reply apps/daimo-mobile/src/view/screen/NoteScreen.tsx:149]: not in this pr ofc. planning to make a reusable send-user-op component that handles (button, simulation/fee display, spinner, success, error state)

File: apps/daimo-mobile/src/view/screen/NoteScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        decimals={tokenMetadata.decimals}
+        displayDecimals={2}
+      />
+      {button}
+      <TextSmall>
+        <TextCenter>{statusMessage}</TextCenter>
+      </TextSmall>
+    </>
+  );
+}
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/NoteScreen.tsx:194]: 🚀

File: apps/daimo-mobile/src/view/screen/SendScreen.tsx [hunk 3]
```diff
@@ -22,23 +36,43 @@ import { TextBody, TextCenter, TextError, TextSmall } from "../shared/text";
...
 
+type SendMethod = "recipient" | "scan" | "createNote" | "unknown";
+
 export default function SendScreen({ route }: Props) {
   const { recipient } = route.params || {};
 
-  const [showScan, setShowScan] = useState(false);
-  const scan = useCallback(() => setShowScan(true), []);
-  const hideScan = useCallback(() => setShowScan(false), []);
+  const [sendMethod, setSendMethod] = useState<SendMethod>("unknown");
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/SendScreen.tsx:44]: 👍 👍

File: apps/daimo-mobile/src/view/screen/SendScreen.tsx [hunk 4]
```diff
@@ -22,23 +36,43 @@ import { TextBody, TextCenter, TextError, TextSmall } from "../shared/text";
...
+            <Search />
+          </>
+        )}
       </ScrollView>
-      {/** TODO: Create Note */}
+      {sendMethod === "unknown" && (
+        <View style={styles.vertCreateNote}>
+          <ButtonBig title="Create Note" onPress={createNote} />
+          <TextSmall>Send via WhatsApp, Signal, TG...</TextSmall>
+        </View>
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/SendScreen.tsx:74]: 🚀

File: apps/daimo-mobile/src/view/screen/SendScreen.tsx [hunk 5]
```diff
@@ -214,6 +254,140 @@ function SendPayment({ recipient }: { recipient: Recipient }) {
...
+          ephemeralOwner,
+          `${dollars}`
+        );
+      }
+    };
+  };
+
+  const { status, message, exec } = useSendAsync(
+    account.enclaveKeyName,
+    createTxHandle(ephemeralPrivateKey, isNotesContractApproved, dollars)
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/SendScreen.tsx:310]: 👍

File: apps/daimo-mobile/src/view/screen/SendScreen.tsx [hunk 6]
```diff
@@ -214,6 +254,140 @@ function SendPayment({ recipient }: { recipient: Recipient }) {
...
+      if (isNotesContractApproved) {
+        console.log(
+          `[ACTION] skipping approve, creating note worth $${dollars}`
+        );
+        return account.createEphemeralNote(ephemeralOwner, `${dollars}`);
+      } else {
+        console.log(
+          `[ACTION] approving notes contract to spend token and creating note worth $${dollars}`
+        );
+        return account.erc20approveAndCreateEphemeralNote(
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/SendScreen.tsx:300]: looks excellent

in future, we might pre-approve both the notes contract & the paymaster in the account constructor

File: apps/daimo-mobile/src/view/shared/Amount.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import { StyleSheet, Text } from "react-native";
+
+import Spacer from "./Spacer";
+import { color } from "./style";
+import { TextH1 } from "./text";
+
+export function TitleAmount({
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/Amount.tsx:7]: 👍 👍

File: packages/daimo-userop/src/index.ts [hunk 8]
```diff
@@ -158,4 +164,81 @@ export class DaimoAccount {
...
+    amount: `${number}`
+  ) {
+    const parsedAmount = parseUnits(amount, this.tokenDecimals);
+    console.log(
+      `[OP] approve ${amount} ${
+        this.tokenAddress
+      } for ${this.getAddress()}, then, create ephemeral note for ${ephemeralOwner} of ${parsedAmount}`
+    );
+
+    const op = this.opBuilder.executeBatch(
```
Comment (dcposch) [inline packages/daimo-userop/src/index.ts:180]: pretty cool benefit of 4337 that you can just do atomic batches natively now

**Review by dcposch — APPROVED** [review]
This looks amazing 🚢

File: apps/daimo-mobile/src/view/screen/NoteScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+import { useAccount } from "../../model/account";
+import { TitleAmount } from "../shared/Amount";
+import { ButtonBig } from "../shared/Button";
+import { HomeStackParamList } from "../shared/nav";
+import { color, ss } from "../shared/style";
+import { TextCenter, TextError, TextSmall } from "../shared/text";
+
+type Props = NativeStackScreenProps<HomeStackParamList, "Note">;
+
+export default function NoteScreen({ route }: Props) {
```
[the remark this answers is not in the document]
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/NoteScreen.tsx]: Done

**Pull request merged**